sherlockml/clients/secret.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2018 ASI Data Science
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from collections import namedtuple
+
+from marshmallow import Schema, fields, post_load
+
+from sherlockml.clients.base import BaseClient
+
+
+DatasetsSecrets = namedtuple(
+    'DatasetsSecrets', ['bucket', 'access_key', 'secret_key', 'verified']
+)
+
+
+class DatasetsSecretsSchema(Schema):
```

Comment:
We could do this, but I'm not confident it would add a lot of extra value, and would complicate the implementation a little. Elsewhere in this library we're not yet doing much validation of this sort, and I would like to err on the side of simpler implementations.